packages/sitecore-jss-dev-tools/src/codegen/utils.ts

```diff
@@ -0,0 +1,113 @@
+import fs from 'fs';
+import * as ts from 'typescript';
+import { constants } from '@sitecore-jss/sitecore-jss';
+
+// TODO:adjust when mesh endpoint is live
+const meshEndpoint = `${process.env.SITECORE_EDGE_URL ||
+  constants.SITECORE_EDGE_URL_DEFAULT}/api/v1/mesh`;
+
+export const resolveComponentImportFiles = (appPath: string) => {
+  appPath = path.isAbsolute(appPath) ? appPath : path.resolve(process.cwd(), appPath);
+  const tsConfig = ts.readConfigFile(path.resolve(appPath, 'tsconfig.json'), ts.sys.readFile);
+
+  if (tsConfig.error) {
+    throw new Error(`Error reading tsconfig.json from JSS app root: ${tsConfig.error.messageText}`);
+  }
+
+  const tsOptions = {
+    ...tsConfig.config.compilerOptions,
+    baseUrl: appPath,
+  };
+
+  const componentBuilderPath = path.resolve(appPath, 'src', 'temp', 'componentBuilder.ts');
+
+  const tsHost = ts.createCompilerHost(tsOptions, true);
+  const sourceFile = tsHost.getSourceFile(componentBuilderPath, ts.ScriptTarget.Latest, (msg) => {
+    throw new Error(`Failed to parse ${componentBuilderPath}: ${msg}`);
+  });
+
+  if (!sourceFile) throw ReferenceError(`Failed to find file ${componentBuilderPath}`);
+
+  const componentImportsMap: Map<string, string> = new Map();
+  const importNodesMap: Record<string, string> = {};
+  let mapExportName: string = '';
+  const mapAssignments: ts.CallExpression[] = [];
+
+  ts.forEachChild(sourceFile, (childNode) => {
+    if (ts.isImportDeclaration(childNode)) {
+      if (childNode.importClause) {
+        const moduleName = childNode.moduleSpecifier.getText().replace(/['"]/g, '');
+        if (!moduleName.startsWith('node:') && moduleName.indexOf('/node_modules') === -1) {
+          const resolvedModule = ts.nodeModuleNameResolver(
+            moduleName,
+            componentBuilderPath,
+            tsOptions,
+            tsHost
+          );
+          const resolvedFile = resolvedModule?.resolvedModule?.resolvedFileName;
+          // module imports will be resolved to /node_modules location - we don't support that yet
+          if (resolvedFile && resolvedFile.indexOf('node_modules') === -1) {
+            // gotta use OS-agnostic path
+            importNodesMap[childNode.importClause.getText()] = path.resolve(resolvedFile);
+          }
+        }
+      }
+    } else if (ts.isExpressionStatement(childNode)) {
+      ts.forEachChild(childNode, (expressionNode) => {
+        if (ts.isCallExpression(expressionNode)) {
+          if (expressionNode.expression.getText().indexOf('set') !== -1) {
+            // get map.set assignments
+            mapAssignments.push(expressionNode);
+          }
+        }
+      });
+    } else if (ts.isExportAssignment(childNode)) {
+      // get component map export variable
+      mapExportName = childNode.expression.getText();
+    }
+  });
+  for (const mapAssignment of mapAssignments) {
+    // only consider exported map
+    if (mapAssignment.getText().startsWith(mapExportName)) {
```

Comment:
Let's reduce nesting here as well
```suggestion
    if (!mapAssignment.getText().startsWith(mapExportName)) {
        continue;
    }
    
    ...
```

Reply:
I agree about other places but in here the change is purely stylistic IMO. 
I would leave the current approach.

Reply:
I don't mind :)